src/ape/managers/networks.py

```diff
@@ -147,31 +190,84 @@ def network_choices(self) -> Iterator[str]:
         Use the CLI command ``ape networks list`` to list all the possible network
         combinations.
 
+        Args:
+            ecosystem_filter (Optional[List[str]]): Get only the specified ecosystems. Defaults
+              to getting all ecosystems.
+            network_filter (Optional[List[str]]): Get only the specified networks. Defaults
+              to getting all networks.
+            provider_filter (Optional[List[str]]): Get only the specified providers. Defaults
+              to getting all providers.
+
         Returns:
             Iterator[str]: An iterator over all the network-choice possibilities.
         """
-        for ecosystem_name, ecosystem in self.ecosystems.items():
-            yield ecosystem_name
-            for network_name, network in ecosystem.networks.items():
-                if ecosystem_name == self.default_ecosystem.name:
-                    yield f":{network_name}"
 
-                yield f"{ecosystem_name}:{network_name}"
+        def _validate_filter(arg: Optional[Union[List[str], str]], options: Set[str]):
```

Comment:
Turn this into a staticmethod or function?

Reply:
> Seems like a lot of work to core to add a feature to `ape_networks`'s CLI

The majority of the work in core is done to support the filtering in `ape.cli.options.network_option()`

Reply:
Alternatively, we can do the filtering for `network_option` within the `ape.cli` namespace.

`ape_networks` is only touching small changes in core.